### PR TITLE
Us9

### PR DESCRIPTION
**Pull request by Airglow98**

### CODE REVIEW

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
Hello @Airglow98,

well done. 🚀  I've left some comments for you. 

🚨**A super important note** 
Upon reviewing your deployment, it appears that the games are not loading as expected. Despite thorough examination of the code, I wasn't able to pinpoint the exact issue. I suggest taking another look to resolve this.


I did notice something about your two toggle functions which might need your attention; however, you may consider revisiting this for refactoring at a later stage.⬇️

- You could think about merging the two toggle functions (`toggleIsWishlist` and `toggleIsLibrary`) into one function  by accepting a key argument. 

Something like this (not proofed  🙂) `toogleGameFunction`:

```js
 function toggleGameStatus (id, key) {
    const updatedGames = games.map((game) => {
      if (game.id === id) {
        const updatedGame = { ...game, [key]: !game[key] };
        updateGameInBackend(id, updatedGame);
        return updatedGame;
      } else {
        return game;
      }
    });

    mutate(updatedGames);
  };
```

**Comment by Airglow98** [comment]
Hey Jessica

thanks for your suggestions. It actually helped me a lot to better understand my own code. 
To explain why I used useEffect: My Idea was, that I need it to render it after data has changed. So if the games(data) are not loaded there is the loading... page. And if the data is loaded (so the data changes) I render data with useEffect. Please let me know if there is a better way to do it. And like you already said I will look for the first suggestion if i do my code refactoring. The rest is already changed. Thanks again 🙌🏻

File: pages/_app.js [hunk 1]
```diff
@@ -1,50 +1,117 @@
...
-  const [games, setGames] = useLocalStorageState("games", {
-    defaultValue: initialGames.map((game) => ({
-      ...game,
-      isLibrary: false,
-      isWishlist: false,
-    })),
-  });
-
-  function toggleIsLibrary(id) {
+  const fetcher = (...args) => fetch(...args).then((res) => res.json());
```
[the remark this answers is not in the document]
Comment (actyvystom) [reply pages/_app.js:7]: `fetcher` is still inside the App component. You can place it directly after your import statements.

File: pages/_app.js [hunk 2]
```diff
@@ -1,50 +1,117 @@
...
+  const { data, isLoading, mutate } = useSWR("/api/games", fetcher);
+  const [games, setGames] = useState([]);
+
+  useEffect(() => {
+    if (data) {
+      setGames(
+        data.map((game) => ({
+          ...game,
+          /*  isLibrary: false,
+          isWishlist: false, in der Datenbank hinzugeügte keys*/
```
[the remark this answers is not in the document]
Comment (actyvystom) [reply pages/_app.js:17]: Commented code is still there...

File: pages/_app.js [hunk 3]
```diff
@@ -1,50 +1,117 @@
...
+        }))
+      );
+    }
+  }, [data]);
```
[the remark this answers is not in the document]
Comment (actyvystom) [reply pages/_app.js:21]: You don't need a useEffect here as `useSWR` would be managing data for you. 

 :exclamation: But your data fetching is redundant as you do another useSWR-call inside your homepage (`index.js`) and you pass it from `_app.js` into this component. 
 Try to refactor your `_app.js` and move the data handling outside of it (keep it in homepage), of course you can keep and pass the state here and pass it down to your homepage.

File: pages/_app.js [hunk 4]
```diff
@@ -1,50 +1,117 @@
...
-        {...pageProps}
-        games={games}
-        onToggleLibraryClick={toggleIsLibrary}
-        onToggleWishlistClick={toggleIsWishlist}
-        setGames={setGames}
-        amountOfLibraryGames={amountOfLibraryGames}
-        amountOfWishlistGames={amountOfWishlistGames}
-      />
+      <SWRConfig
+        value={{
```
Comment (actyvystom) [inline pages/_app.js:93]: Once you define fetcher outside of your app component, you don`t need to define it again here. Just pass it in like:
```suggestion
        value={{fetcher}}
```

**Review by actyvystom — CHANGES_REQUESTED** [review]
I hope I didn't get your changes wrong (compared it also in your current branch outside of this PR). See some comments on your usage of useSWR and it's data in general.

Cheers, Thomas

**Pull request merged**